packages-exp/messaging-exp/src/helpers/register.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  ComponentContainer,
+  ComponentType,
+  InstanceFactory
+} from '@firebase/component';
+import { ERROR_FACTORY, ErrorCode } from '../util/errors';
+
+import { MessagingService } from '../messaging-service';
+import { _registerComponent } from '@firebase/app-exp';
+import { isSupported } from './isSupported';
+
+const messagingFactory: InstanceFactory<'messaging-exp'> = (
```

Comment:
I see a CI build error:

```
[!] (plugin rpt2) Error: /home/runner/work/firebase-js-sdk/firebase-js-sdk/packages-exp/messaging-exp/src/helpers/register.ts(30,41): semantic error TS2344: Type '"messaging-exp"' does not satisfy the constraint 'never'.
src/helpers/register.ts
Error: /home/runner/work/firebase-js-sdk/firebase-js-sdk/packages-exp/messaging-exp/src/helpers/register.ts(30,41): semantic error TS2344: Type '"messaging-exp"' does not satisfy the constraint 'never'.
```

I'm not totally sure how it works, I see it is declared in `messaging-types-exp` but this file doesn't seem to know about it. Maybe it needs to import `@firebase/messaging-types-exp`? In storage-exp I declared it right inside index.ts:

https://github.com/firebase/firebase-js-sdk/blob/9fd3f5233077b45c5101789c427db51835484ce0/packages/storage/exp/index.ts#L105